tools/deploy.py

```diff
@@ -53,17 +49,36 @@ def deploy_pew_pew_control_module(containerPath):
 def deploy_py3status_module():
     configHomePath = Path(os.getenv("XDG_CONFIG_HOME", "~")).expanduser()
     candidates = [
-        configHomePath / ".config/py3status/modules",
-        configHomePath / ".config/i3status/py3status",
-        configHomePath / ".config/i3/py3status",
-        configHomePath / ".i3/py3status",
+        configHomePath / ".config" / "py3status" / "modules",
+        configHomePath / ".config" / "i3status" / "py3status",
+        configHomePath / ".config" / "i3" / "py3status",
+        configHomePath / ".i3" / "py3status",
     ]
     path = first_existing(candidates)
     if path is not None:
         log.info(f"deploy py3status module {PY3STATUS_MODULE_PATH} to {path}")
         shutil.copy(PY3STATUS_MODULE_PATH, path)
     else:
-        raise Exception("Could not find a home for pewpew")
+        raise Exception("Could not find a home for {}".format(MODULE_NAME))
+
+
+def warn_if_user_not_in_expected_groups():
+    user = getpass.getuser()
+    groups = [g.gr_name for g in grp.getgrall() if user in g.gr_mem]
+    expected_groups = ['input', 'dialout']
```

Comment:
for Arch I can already say that this needs to be uucp instead of dialout. I also don't need to be in input.

Reply:
OK, so checking if it's Ubuntu or Arch is one approach. Is there a Pythonic/robust way of checking distribution?

Or, as you suggest actually testing for the operation we need to be able to do, and then instead just print a "hint" about these things if the check fails. It seems like the more robust/long term solution...

Reply:
For the 'dialout'/'uucp' check there is package for checking what distribution we're on: https://pypi.org/project/distro/

The 'input' group is Ubuntu specific, but could also be handled by distro checking.

The general problem of checking for if evdev is possible for current user, and USB-serial comm (dialout/uucp group) would be tricker.

Both of those dependencies (pyserial, evdev) need to be installed in the environment in which py3status is executing BTW, guess that would be part of the PyPI py3status-pewpew package?

Reply:
I also think that "feature" testing is the most robust approach, but playing with distro for now and collecting different needs for different distros might also not hurt. I'd say go with whatever you fancy for now.

> BTW both of those dependencies (pyserial, evdev) need to be installed in the environment in which py3status is executing, guess that would be part of the PyPI py3status-pewpew package?

Right - we should also document that properly as @ultrabug suggested we use this package as showcase for the new feature. Will open a bug for that.

Reply:
OK I'll make an attempt with distro and the knowledge about ubuntu/arch we have so far, then I'll hope this PR passes the gates ;)

Reply:
Added the distro check and logic.
Also, reordered functions top-down as per the main-at-top philosophy. It might make sense to review the complete view of deploy.py script instead of diff (it's becoming unreadable).
Also, took the liberty to refactor main() function slightly, again for readability.
Hopy you like it!